packages/asset-swapper/src/utils/rfq_maker_blacklist.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * Tracks a maker's history of timely responses, and manages whether a given
+ * maker should be avoided for being too latent.
+ */
+export class RfqMakerBlacklist {
+    // tslint:disable-next-line:custom-no-magic-numbers
+    private static readonly _msPerMinute = 1000 * 60;
+    private readonly _makerTimeoutStreakLength: { [makerUrl: string]: number } = {};
+    private readonly _makerBlacklistedUntilDate: { [makerUrl: string]: number } = {};
+    constructor(private readonly _blacklistDurationMinutes: number, private readonly _timeoutStreakThreshold: number) {}
+    public logTimeoutOrLackThereof(makerUrl: string, didTimeout: boolean): void {
+        if (!this._makerTimeoutStreakLength.hasOwnProperty(makerUrl)) {
+            this._makerTimeoutStreakLength[makerUrl] = 0;
+        }
+        if (didTimeout) {
+            this._makerTimeoutStreakLength[makerUrl] += 1;
```

Comment:
`this._makerTimeoutStreakLength[makerUrl] = (this._makerTimeoutStreakLength[makerUrl] || 0) + 1` could allow you to omit L12-14

Reply:
I loved the suggestion at first.  But then, when I wrote it up, I decided that I prefer to keep the simple `...Streak... += 1;`, since that's such a key part of the mechanism here, and I'd like to keep it as simple and straightforward as possible.